### PR TITLE
Subscription

### PR DESCRIPTION
**Pull request by neera11**
It is new separate plugin for subscription button 
Please review and approve

### CODE REVIEW

**Review by ChetanGN — CHANGES_REQUESTED** [review]
@neera11 Add .DS_Store file to git ignore.

File: README.md [hunk 1]
```diff
@@ -1,2 +0,0 @@
-# subscription-button-wordpress-plugin
-Subscription button plugin for wordpress.
```
Comment (ChetanGN) [inline README.md:2]: Add readme  content.

Comment (neera11) [reply README.md:2]: added in readme.txt file

Comment (ChetanGN) [reply README.md:2]: Then please remove this file.

File: includes/errors/payment-button-error-code.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Razorpay\PaymentButton\Errors;
+
+use Razorpay\Api\Errors as ApiErrors;
+
+class Payment_Button_Error_Code extends ApiErrors\ErrorCode
+{
+    // Razorpay Payment Button
+    const API_PAYMENT_BUTTON_FETCH_FAILED     = 'Razorpay payment button fetch request failed';
```
Comment (ChetanGN) [inline includes/errors/payment-button-error-code.php]: @neera11 is it a payment button or a subscription button? Please change the error message according to that only.

File: readme.txt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+1. Install the plugin from the [Wordpress Plugin Directory].
+2. Navigate to Razorpay Subscription Buttons → Settings to enter configuration details of Razorpay.
+3. Navigate to Razorpay Subscription Buttons and check the list of buttons available on Razorpay.
+4. Navigate to Pages → All Pages/Add New to embed Payment button to WordPress page and select page to embed button on this.
+5. Select Add Block → Razorpay: Subscription Button widget to embed Payment button to page/posts.
+6. Choose Subscription Button using the drop down button.
+
+== Changelog ==
+
+
```
Comment (ChetanGN) [inline readme.txt]: Remove extra space here.

File: release.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+#
+# Source: https://gist.github.com/philbuchanan/8188898
+# License: GNU GPL 2.1
+# Script to deploy from Github to WordPress.org Plugin Repository
+# A modification of Dean Clatworthy's deploy script as found here: https://github.com/deanc/wordpress-plugin-git-svn
+# The difference is that this script lives in the plugin's git repo & doesn't require an existing SVN repo.
+# Slightly adapted
+
+SVNUSER=razorpay
+PLUGINSLUG='razorpay-payment-button'
```
Comment (ChetanGN) [inline release.sh]: Change the slug name here.

File: includes/errors/payment-button-error-code.php (hunk 2 above)
Comment (neera11) [reply includes/errors/payment-button-error-code.php]: changed

File: release.sh (hunk 4 above)
Comment (neera11) [reply release.sh]: slug name will get from wordpress.org

Comment (ChetanGN) [reply release.sh]: Wordpress.org will pick the name from the plugin header. So you can name the slug according to that. @neera11

File: includes/errors/payment-button-error-code.php [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Razorpay\PaymentButton\Errors;
+
+use Razorpay\Api\Errors as ApiErrors;
+
+class Payment_Button_Subscription_Error_Code extends ApiErrors\ErrorCode
+{
+    // Razorpay Payment Button
+    const API_PAYMENT_BUTTON_FETCH_FAILED     = 'Razorpay subscription button fetch request failed';
```
Comment (ChetanGN) [inline includes/errors/payment-button-error-code.php]: @neera11 Change the constant name also.

File: includes/errors/payment-button-error-code.php [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Razorpay\PaymentButton\Errors;
```
Comment (ChetanGN) [inline includes/errors/payment-button-error-code.php]: Change the namespace here.

File: includes/errors/payment-button-error-code.php (hunk 5 above)
Comment (neera11) [reply includes/errors/payment-button-error-code.php]: changed

File: includes/errors/payment-button-error-code.php (hunk 6 above)
Comment (neera11) [reply includes/errors/payment-button-error-code.php]: changed

File: release.sh (hunk 4 above)
Comment (neera11) [reply release.sh]: changed

**Review by ChetanGN — APPROVED** [review]
LGTM.

**Pull request merged**